internal/codespaces/api/api.go

```diff
@@ -186,16 +187,42 @@ type CodespaceEnvironmentConnection struct {
 	HostPublicKeys []string `json:"hostPublicKeys"`
 }
 
+// codespacesListResponse is the request body for the `/user/codespaces` endpoint
+type getCodespacesListResponse struct {
+	Codespaces []*Codespace `json:"codespaces"`
+	TotalCount int          `json:"total_count"`
+}
+
 // ListCodespaces returns a list of codespaces for the user.
-func (a *API) ListCodespaces(ctx context.Context) ([]*Codespace, error) {
+func (a *API) ListCodespaces(ctx context.Context) (codespaces []*Codespace, err error) {
+	for page := 1; ; page++ {
+		response, err := a.fetchCodespaces(ctx, page)
+		if err != nil {
+			return nil, fmt.Errorf("%w", err)
+		}
+		if len(codespaces) >= response.TotalCount || len(response.Codespaces) == 0 {
```

Comment:
Usually the way I do this with pagination is something like this:
```
if page * per_page >= response.TotalCount // we've reached the end
```

So if the total count is 101, at 50 per page, we have 3 pages, on page 3, it would `3 * 50 = 150` which is greater than 101. This removes the need to check the actual codespaces we've fetched.